src/test/groovy/org/powerflows/dmn/domain/model/DecisionSpec.groovy

```diff
@@ -0,0 +1,245 @@
+                        .value(someInput1Expression1Value)
+                        .and()
+                    .next()
+                    .name(someInput2Name)
+                    .description(someInput2Description)
+                    .type(someInput2Type)
+                    .end()
+                .withOutputs()
+                    .name(someOutput1Name)
+                    .description(someOutput1Description)
+                    .type(someOutput1Type)
+                    .withExpression()
+                        .type(someOutput1Expression1Type)
+                        .value(someOutput1Expression1Value)
+                        .and()
+                    .next()
+                    .name(someOutput2Name)
+                    .description(someOutput2Description)
+                    .type(someOutput2Type)
+                    .end()
+                .withRules()
+                    .description(someRule1Description)
+                    .withInputEntries()
+                        .name(someRule1InputEntry1Name)
+                        .withExpression()
+                            .type(someRule1InputEntry1ExpressionType)
+                            .value(someRule1InputEntry1ExpressionValue)
+                            .and()
+                        .next()
+                        .name(someRule1InputEntry2Name)
+                        .withExpression()
+                            .type(someRule1InputEntry2ExpressionType)
+                            .value(someRule1InputEntry2ExpressionValue)
+                            .and()
+                        .end()
+                    .withOutputEntries()
+                        .name(someRule1OutputEntry1Name)
+                        .withExpression()
+                            .type(someRule1OutputEntry1ExpressionType)
+                            .value(someRule1OutputEntry1ExpressionValue)
+                            .and()
+                        .end()
+                    .next()
+                    .description(someRule2Description)
+                    .withInputEntries()
+                        .name(someRule2InputEntry1Name)
+                        .end()
+                    .withInputEntries()
+                        .name(someRule2InputEntry2Name)
+                        .end()
+                    .withOutputEntries()
+                        .name(someRule2OutputEntry1Name)
+                        .next()
+                        .name(someRule2OutputEntry2Name)
+                        .end()
+                    .end()
+                .build()
+        // @formatter:on
+
+        then:
+        decision != null
+
+        with(decision) {
+            getId() == someTableId
+            getName() == someTableName
+            getHitPolicy() == someHitPolicy
+            getInputs().size() == 2
+            getOutputs().size() == 2
+            getRules().size() == 2
+        }
+
+        final Input input1 = decision.getInputs().get(0)
```

Comment:
this variable is used only once... the same applies to the ones below

Reply:
In this way is more clean in my opinion.